WooCommerce/Classes/ViewRelated/ReusableViews/EnhancedTextView.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+
+/// A text view which support the placeholder label.
+///
+final class EnhancedTextView: UITextView {
+
+    var onTextChange: ((String) -> Void)?
+
+    var placeholder: String? {
+        didSet {
+            placeholderLabel?.text = placeholder
+            placeholderLabel?.sizeToFit()
+        }
+    }
+    private var placeholderLabel: UILabel?
+
+    override var text: String! {
+        didSet {
+            if text.isEmpty {
+                animatePlaceholder()
+            }
+            else {
+                hidePlaceholder()
+            }
+        }
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        delegate = self
+        placeholderLabel = UILabel(frame: self.bounds)
+        configureLabels()
+        if let unwrappedLabel = placeholderLabel {
+            addSubview(unwrappedLabel)
+        }
```

Comment:
nit: maybe the above `placeholderLabel = UILabel(frame: self.bounds)` line and this `addSubview` could be grouped into a `configure` function like `configurePlaceholderLabel`?